### ISSUE TITLE
Double buffering issue on Android?

### ISSUE THREAD

**Issue by Rippletank** [mono/SkiaSharp#681]
### Description

There appears to be an issue with double-buffering on android. When items are drawn within the OnPaintSurface method, as I understand, they should only be rendered once the method has finished. On IOS and UWP, this appears to be the case but not on Android.

### Code

The attached example reproduces the problem. It is a fresh xamarin project with an added class based on SKGLView as the MainForm content. It renders two white "cards" with code (not shown here) to allow swiping the cards left and right.

Here is the OnPaintSurface method:

        protected override void OnPaintSurface(SKPaintGLSurfaceEventArgs e)
        {
            var canvas = e.Surface.Canvas;
            canvas.Clear(Color.DarkBlue.ToSKColor());

            float h = e.RenderTarget.Height;
            float w = e.RenderTarget.Width;

            //Left Card
            SimulatedCardPaint(canvas, -AnimationPosition, w, h);

            //Right Card
            SimulatedCardPaint(canvas, 1 - AnimationPosition, w, h);
        }

        public void SimulatedCardPaint(SKCanvas canvas, float TransitionPosition, float w, float h)
        {
            canvas.ResetMatrix();
            canvas.Translate(TransitionPosition * w, 0);

            using (var paint = new SKPaint()
            {
                Style = SKPaintStyle.Fill,
                Color = Color.White.ToSKColor(),
            })
            {
                var rect = new SKRect(0, 0, w, h);
                rect.Inflate(-20, -20);
                canvas.DrawRect(rect, paint);
            }
            Task.Delay(20).Wait();
        }

The delay simulates a more complex drawing operation. The original code had no async methods at all but still had the same results.

### Expected Behavior

As you swipe left and right with your finger on the screen, the "cards" should move with a dark blue border line between them. The line should always be the same width because the cards are both rendered inside a single call to OnPaintSurface.

### Actual Behavior

On IOS and UWP, the blue line appears to be the same size all the time. 
On Android, the line can be made bigger and smaller depending on the speed of the swipe and there are occasional flickers which appear to be the background being rendered without the cards. Neither of these should be possible as the cards are rendered in the same method. The only way I can see this occurring is if the canvas is rendered in the middle of the drawing. It gives the drawing a flacky feel unlike the solid feel of the UWP and IOS drawing.

### Basic Information

This was based on testing on hardware, not on simulators. The android hardware is the most modern so should not have a performance issue at all.

This is using version 1.60.3 of SkiaSharp.

### Screenshots

If necessary, I could provide photographs or video but it is hard to get screenshots as it only happens during motion.

### Reproduction Link

[SkiaDoubleBufferTest.zip](https://github.com/mono/SkiaSharp/files/2571718/SkiaDoubleBufferTest.zip)

**Comment by Rippletank** [comment]
Just stumbled upon the problem.

Looks like the issue is that on Android OnTouch and OnPaintSurface calls can overlap where as on UWP they don't appear to do so at all. I haven't had chance to confirm this with IOS yet.

So interesting lesson is don't assume touch events will not happen during in the middle of a paint operation,

**Comment by Rippletank** [comment]
Thanks, that confirms what I suspected (you might have missed my comment from couple of days ago).

It's useful to know. Thinking about it, I'm pretty sure it affects xamarin.forms in general. One reason I switched to using skiasharp was because after a lot of preliminary work on a recent app using uwp and ios, the android forms version had similar problems. I thought it was performance related. Not so sure now.

Anyway, Skiasharp is fantastic so thanks for everything.
